### ISSUE TITLE
Resizable SplitPageLayout is not discoverable for non-screen reader users

### ISSUE THREAD

**Issue by jdrush89** [primer/react#3006]
### Description

I don't believe the solution to https://github.com/primer/react/issues/2522 works for all users. In https://github.com/primer/react/pull/2593 a hidden focusable input was added to control the width of the SplitPageLayout panes. The purpose of the input is announced to screen reader users, but non-screen reader users have no idea that it is there or what it is for. It's just adding an extra tab stop for them with no visual indication which is confusing. The pane needs to be resizable for keyboard only users who are not using a screen reader as part of [WCAG 2.1](https://www.w3.org/WAI/WCAG21/Understanding/keyboard-accessible). A better solution would be to use the [aria separator role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/separator_role#focusable_separator) with accompanying attributes. An [implementation](https://azdevinternal.azureedge.net/components/splitter) of this that works for both screen reader and non-screen reader users can be seen in the ADO component library.

### Steps to reproduce

1. Go to https://primer.style/react/SplitPageLayout#with-resizable-pane
2. Try tabbing to the hidden input and changing the pane width without a screen reader. There's no indication of where the hidden input is, what values are allowed, or that you have to type a number and hit enter.

### Version

v35.20

### Browser

Chrome

**Comment by broccolinisoup** [comment]
Just linking this issue here that might be helpful for extra context. https://github.com/github/primer/issues/1865

**Comment by lesliecdubs** [comment]
Thanks for filing @jdrush89! We're actively working on the issue @broccolinisoup linked above, which may help resolve this. If it's not resolved by that issue, we'll keep this in the backlog to pick up after nearer-term accessibility efforts with broader reach (particularly https://github.com/github/primer/issues/1627 & https://github.com/github/primer/issues/1790) have been addressed.

**Comment by lesliecdubs** [comment]
Noting that this issue is blocking completion of an a11y audit issue, which I didn't notice when reviewing it earlier. https://github.com/github/accessibility-audits/issues/3477

**Comment by lesliecdubs** [comment]
Not stale. 

I believe there is active exploration happening in https://github.com/primer/react/pull/3903 that might address this; @pksjce would you mind taking a look at the issue described/linked here and see if you can validate whether the splitter approach would resolve this?

**Comment by pksjce** [comment]
Hi @jdrush89 - Thanks for filing this issue. I have been working on making this splitter more accessible and implemented the `role=separator` as seen on the ADO component library. Unfortunately we got a11y feedback that JAWS + Firefox which is a popular combination is unable to read the respective aria values. I have updated my PR with new guidance and I am awaiting feedback. Will close this issue once my PR gets merged.

**Comment by TylerJDev** [comment]
Hey @pksjce! With your changes merged in #3903, should we close this out? If so, we can probably ask to get this issue closed out too https://github.com/github/accessibility-audits/issues/3477 :grin:

**Comment by pksjce** [comment]
Yes! This issue is now fixed.
